src/ui/Base/RegisteredNamedMethods.ts

```diff
@@ -447,6 +448,68 @@ Initialization.registerNamedMethod('configureResourceRoot', (path: string) => {
   configureResourceRoot(path);
 });
 
+function findAllComponentsOfType<T extends BaseComponent>(type: { new (...args: any[]): T }, searchRoot = document.body) {
+  const className = Component.computeCssClassNameForType((<typeof BaseComponent>(<{}>type)).ID);
+  const root = $$(searchRoot);
+  const found: T[] = [];
+  if (root.hasClass(className)) {
+    found.push(<T>Component.resolveBinding(root.el, type));
+  }
+  found.push(...root.findClass(className).map(element => <T>Component.resolveBinding(element as HTMLElement, type)));
+  return found;
+}
+
+/**
+ * Re-enables all [Analytics]{@link Analytics} components if they were previously disabled.
+ * @param searchRoot What ancestor or CoveoAnalytics element to search from in the DOM.
+ */
+export function enableAnalytics(searchRoot = document.body) {
+  const analyticsInstances = findAllComponentsOfType(Analytics, searchRoot);
+  if (analyticsInstances.length === 0) {
+    console.warn('Could not enable analytics because they could not be found.');
+    return;
+  }
+  analyticsInstances.forEach(analytics => analytics.enable());
+}
+
+Initialization.registerNamedMethod('enableAnalytics', () => {
+  enableAnalytics();
+});
+
+/**
+ * Removes the analytics tracking cookie and clears the actions history.
+ * @param searchRoot What ancestor or CoveoAnalytics element to search from in the DOM.
+ */
+export function clearLocalData(searchRoot = document.body) {
+  const analyticsInstances = findAllComponentsOfType(Analytics, searchRoot);
+  if (analyticsInstances.length === 0) {
+    console.warn('Could not clear local data because analytics could not be found.');
+    return;
+  }
+  analyticsInstances.forEach(analytics => analytics.clearLocalData());
+}
+
+Initialization.registerNamedMethod('clearLocalData', () => {
+  clearLocalData();
+});
+
+/**
+ * Disables all [Analytics]{@link Analytics} components and clears local data.
+ * @param searchRoot What ancestor or CoveoAnalytics element to search from in the DOM.
+ */
+export function disableAnalytics(searchRoot = document.body) {
+  const analyticsInstances = findAllComponentsOfType(Analytics, searchRoot);
+  if (analyticsInstances.length === 0) {
+    console.warn('Could not disable analytics because they could not be found.');
```

Comment:
Same about logger
Since you have this piece of code with the same error message in multiple places you can extract it to a separate method. E.g a getter `get analyticsInstances()`